base_layer/key_manager/src/mnemonic.rs

```diff
@@ -100,7 +105,8 @@ impl MnemonicLanguage {
                 return Err(MnemonicError::UnknownLanguage);
             },
             Ordering::Greater => {
-                for word in words {
+                for word_ind in 0..words.len() {
```

Comment:
Is it worth it to have `SeedWords` implement an iterator that provides word references? Then you don't need to mess around with indexes.

Reply:
Ah, I hadn't considered that lifetimes would be problematic when writing this. As you say, it's probably not worth it for the two times when it would be used, and the bounds checks are correct as implemented anyway.